convert2rhel/unit_tests/actions/is_loaded_kernel_latest_test.py

```diff
@@ -0,0 +1,579 @@
+                        "--setopt=reposdir=%s" % fake_reposdir_path,
+                        package_name,
+                    ),
+                    (
+                        repoquery_version,
+                        return_code,
+                    ),
+                ),
+                (("uname", "-r"), (uname_version, return_code)),
+            ),
+        )
+        monkeypatch.setattr(
+            actions.is_loaded_kernel_latest,
+            "run_subprocess",
+            value=run_subprocess_mocked,
+        )
+
+        is_loaded_kernel_latest_action.run()
+
+        repoquery_kernel_version = repoquery_version.split("\t")[2]
+        uname_kernel_version = uname_version.rsplit(".", 1)[0]
+        assert is_loaded_kernel_latest_action.message == expected
+        unit_tests.assert_actions_result(
+            is_loaded_kernel_latest_action,
+            status="ERROR",
+            error_id="INVALID_KERNEL_PACKAGE",
+            message=expected,
+        )
+
+    @centos8
+    def test_is_loaded_kernel_latest_eus_system(
+        self, pretend_os, tmpdir, monkeypatch, caplog, is_loaded_kernel_latest_action
+    ):
+        fake_reposdir_path = str(tmpdir)
+        monkeypatch.setattr(
+            actions.is_loaded_kernel_latest,
+            "get_hardcoded_repofiles_dir",
+            value=lambda: fake_reposdir_path,
+        )
+
+        monkeypatch.setattr(actions.is_loaded_kernel_latest.system_info, "has_internet_access", True)
+
+        run_subprocess_mocked = mock.Mock(
+            spec=run_subprocess,
+            side_effect=run_subprocess_side_effect(
+                (
+                    (
+                        "repoquery",
+                        "--setopt=exclude=",
+                        "--quiet",
+                        "--qf",
+                        "C2R\\t%{BUILDTIME}\\t%{VERSION}-%{RELEASE}\\t%{REPOID}",
+                        "--setopt=reposdir=%s" % fake_reposdir_path,
+                        "kernel-core",
+                    ),
+                    (
+                        "C2R\t1634146676\t3.10.0-1160.45.1.el7\tbaseos",
+                        0,
+                    ),
+                ),
+                (("uname", "-r"), ("3.10.0-1160.45.1.el7.x86_64", 0)),
+            ),
+        )
+        monkeypatch.setattr(
+            actions.is_loaded_kernel_latest,
+            "run_subprocess",
+            value=run_subprocess_mocked,
+        )
+
+        is_loaded_kernel_latest_action.run()
+        assert "The currently loaded kernel is at the latest version." in caplog.records[-1].message
+
+    @centos8
+    def test_is_loaded_kernel_latest_eus_system_no_connection(
+        self, pretend_os, monkeypatch, tmpdir, caplog, is_loaded_kernel_latest_action
+    ):
+        monkeypatch.setattr(actions.is_loaded_kernel_latest, "get_hardcoded_repofiles_dir", value=lambda: str(tmpdir))
+        system_info.has_internet_access = False
```

Comment:
Let's use `monkeypatch` here to not cause hidden problems with other unit tests by mistake